.circleci/config.yml

```diff
@@ -0,0 +1,92 @@
+defaults: &defaults
+  working_directory: ~/artsy/reaction
+  docker:
+    - image: circleci/node:8
+  
+save_cache: &save_cache
+  save_cache:
+    key: yarn-deps-{{ checksum "yarn.lock" }}
+    paths:
+      - ./node_modules
+
+restore_cache: &restore_cache
+  restore_cache:
+    keys:
+    - yarn-deps-{{ checksum "yarn.lock" }}
+    - yarn-deps-
+      
+verify_dependencies: &verify_dependencies
+  run: yarn --check-files
+  
+filter_master: &filter_master
+  filters:
+    branches:
+      only:
+        - master
+
+version: 2
+jobs:
+  relay:
+    <<: *defaults
+    docker:
+    - image: circleci/node:8
+    steps:
+      - checkout
+      - <<: *restore_cache
+      - <<: *verify_dependencies
+      - run: yarn relay
+  lint:
+    <<: *defaults
+    steps: 
+      - checkout
+      - <<: *restore_cache
+      - <<: *verify_dependencies
+      - run: yarn lint 
+  type-check:
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *restore_cache
+      - <<: *verify_dependencies
+      - run: yarn type-check
+  deploy:
+    <<: *defaults
+    steps: 
+      - checkout
+      - <<: *restore_cache
+      - <<: *verify_dependencies
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> ~/.npmrc
+      - run: yarn semantic-release
+      - run: cp .env.oss .env
+      - run: yarn deploy-storybook
+  test:
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *restore_cache
+      - <<: *verify_dependencies
+      - run: yarn test -w 4
+  update-cache:
+    <<: *defaults
+    steps:
+      - checkout
+      - <<: *restore_cache
+      - <<: *verify_dependencies
+      - <<: *save_cache
+workflows:
+  version: 2
+  build_and_verify:
+    jobs:
+      - update-cache
+      - relay
+      - lint
+      - type-check
+      - test
```

Comment:
It's worth noting again that as we move more projects to Circle Workflows, we might need to optimize the container use (or increase the number of containers) to avoid PRs blocking each other.

Open source projects got 3 more containers, so the parallel jobs here will take 2 containers from our org. I am not super worried here because each of the jobs will finish in seconds, but something to watch for more expensive jobs from other projects.

Reply:
If there are multiple long lived jobs, you definitely would want to gate on the shorter ones and have those fail before kicking off a longer process. 

According to the circle config we have 7 containers. It's not clear to me if that's per project or if it's for the entire organization.

Reply:
The 7 containers are shared among all projects in the organization, and you can observe the queues at https://circleci.com/gh/artsy.

Our current approach is that expensive builds like Volt and Gravity are configured to take 6 containers and that opens up at least 1 container for smaller builds.

Reply:
Well, looks like we actually have 8 containers (7 paid) because the first one is free. Open source projects get 3 more, so that's why it showed Reaction can have up to 11.